examples/graphql-spring-boot/src/test/kotlin/org/neo4j/graphql/examples/graphqlspringboot/controller/QueriesTest.kt

```diff
@@ -0,0 +1,120 @@
+
+import com.expediagroup.graphql.types.GraphQLRequest
+import org.junit.jupiter.api.Test
+import org.neo4j.graphql.examples.graphqlspringboot.config.Neo4jProperties
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.boot.autoconfigure.EnableAutoConfiguration
+import org.springframework.boot.context.properties.ConfigurationProperties
+import org.springframework.boot.test.context.SpringBootTest
+import org.springframework.boot.test.context.TestConfiguration
+import org.springframework.context.annotation.Bean
+import org.springframework.context.annotation.Primary
+import org.springframework.http.MediaType
+import org.springframework.test.web.reactive.server.WebTestClient
+import org.testcontainers.containers.Neo4jContainer
+import org.testcontainers.junit.jupiter.Container
+import org.testcontainers.junit.jupiter.Testcontainers
+import java.net.URI
+
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+@EnableAutoConfiguration
+@Testcontainers
+internal class QueriesTest(
+        @Autowired private val testClient: WebTestClient
+) {
+    @Test
+    fun `test additional method`() {
+        assertQuery(
+                """
+                    {
+                        echo(string: "hello world")
+                    }
+                """.trimIndent(),
+                """
+                {
+                    "data": {
+                        "echo": "hello world"
+                    }
+                }
+                """.trimIndent())
+    }
+
+    @Test
+    fun `test additional method with pojo`() {
+        assertQuery(
+                """
+                    {
+                        pojo(param: "hello world") {
+                            id
+                        }
+                    }
+                """.trimIndent(),
+                """
+                {
+                    "data": {
+                        "pojo": {
+                            "id": "hello world"
+                        }
+                    }
+                }
+                """.trimIndent())
+    }
+
+    @Test
+    fun `test neo4jQuery`() {
+        assertQuery(
+                """
+                mutation {
+                  createTeam(id: "team1", name: "Team 1") {
+                    id
+                  }
+                }
+                """.trimIndent(),
+                """
+                {
+                    "data": {
+                        "createTeam": { "id": "team1" }
+                    }
+                }
+                """.trimIndent())
+        assertQuery("{ team { id } }",
+                """
+                {
+                    "data": {
+                        "team": [ { "id": "team1" } ]
+                    }
+                }
+                """.trimIndent())
+    }
+
+    private fun assertQuery(query: String, expectedResult: String) {
+        val request = GraphQLRequest(query = query)
+        testClient.post()
```

Comment:
would it make sense to use the graphql client libray as a test client?

Reply:
I don't think we need an additional lib just to trigger the endpoint.